### ISSUE TITLE
Very long lines in code fences cannot be wrapped

### ISSUE THREAD

**Issue by justinsteven** [omnivore-app/omnivore#3383]
I am using https://minimaxir.com/2023/12/chatgpt-structured-data/ as a test case

The original HTML has things such as:

```html
<code class=language-plaintext data-lang=plaintext><span class=line><span class=cl>Bullet Points:
</span></span><span class=line><span class=cl>- Utilize the paid ChatGPT API for more control over output and to maintain a given JSON schema
</span></span><span class=line><span class=cl>- Learn about prompt engineering and system prompts to improve ChatGPT&#39;s output quality
</span></span><span class=line><span class=cl>- Simplify schema input/output with Pydantic for structured data support
</span></span><span class=line><span class=cl>
</span></span><span class=line><span class=cl>Blurb for Social Media:
</span></span><span class=line><span class=cl>Discover how to unlock the full potential of ChatGPT with the paid API, prompt engineering, and Pydantic for structured data support. Gain more control over output and learn to maintain a given JSON schema. #ChatGPT #AI #Pydantic

[... SNIP ...]
```

This renders in the browser (per the author's styling) as:

![image](https://github.com/omnivore-app/omnivore/assets/1893909/110d9fb3-6524-4b70-8874-2aa8ff78279c)

(The 'copy' button appears on hover)

i.e. the author chose to put very long lines in a `<code>` block knowing that their website causes it to be rendered with wrapped lines

Furthermore, when reading source code with very long lines, some readers may prefer that code is not wrapped. Others may prefer for very long lines to be wrapped. This may be helpful for them when horizontal space is limited (e.g. mobile devices).

When rendered in the Omnivore Web UI, the above shows as:

![image](https://github.com/omnivore-app/omnivore/assets/1893909/b5c57dcc-541e-4165-b89b-e1518c3671bf)

Upon mousing over the block, my browser shows a horizontal scroll bar.

It renders similarly on the iOS and iPadOS apps. In these cases, the user must swipe horizontally within the code block for the scrollbar to appear and for it to be evident that there is additional content. In my experience, swiping back and forth to read overly long lines in a code block makes it very easy to trigger #2880

Suggested fixes:

* Be opinionated that code fences should not have wrapped lines, and authors should not be putting overly large lines in them expecting them to be wrapped; OR
* Always wrap long lines in code fences (this may upset people who prefer code blocks to not be wrapped); OR
* Add a display setting that toggles line wrapping in code fences; OR
* Add a per-code-fence setting that toggles line wrapping

**Comment by justinsteven** [comment]
Looks like this was fixed in #3359 (perhaps by accident, I know @jacksonh was locally hacking on this when I brought it up on Discord and it might have accidentally leaked in to a commit). If this change sticks, this issue can be closed

**Comment by jacksonh** [comment]
Thanks for finding this ticket! I will close it if you think its all good
